rotkehlchen/chain/ethereum/modules/thegraph/decoder.py

```diff
@@ -0,0 +1,187 @@
+# example delegate() call: https://etherscan.io/tx/0x6ed3377db652151fb8e4794dd994a921a2d029ad317bd3f2a2916af239490fec
+TOPIC_STAKE_DELEGATED = b'\xcd\x03f\xdc\xe5$}\x87O\xfc`\xa7b\xaaz\xbb\xb8,\x16\x95\xbb\xb1q`\x9c\x1b\x88a\xe2y\xebs'  # noqa: E501
+# example undelegate() call: https://etherscan.io/tx/0x5ca5244868d9c0d8c30a1cad0feaf137bd28acd9c3f669a09a3a199fd75ad25a
+TOPIC_STAKE_DELEGATED_LOCKED = b'\x040\x18?\x84\xd9\xc4P#\x86\xd4\x99\xda\x80eC\xde\xe1\xd9\xde\x83\xc0\x8b\x01\xe3\x9am!\x16\xc4;%'  # noqa: E501
+# example withdrawDelegated() call: https://etherscan.io/tx/0x49307751de5ba4cf98fccbdd1ab8387fd60a7ce120800212c216bf0a6a04acfa
+TOPIC_STAKE_DELEGATED_WITHDRAWN = b'\x1b.w7\xe0C\xc5\xcf\x1bX|\xebM\xae\xb7\xae\x00\x14\x8b\x9b\xda\x8fy\xf1\t>\xea\xd0\x8f\x14\x19R'  # noqa: E501
+
+
+class ThegraphDecoder(DecoderInterface):
+    def __init__(
+            self,
+            ethereum_inquirer: 'EthereumInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+    ) -> None:
+        super().__init__(
+            evm_inquirer=ethereum_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+        self.token = A_GRT.resolve_to_evm_token()
+
+    def counterparties(self) -> list[CounterpartyDetails]:
+        return [THEGRAPH_CPT_DETAILS]
+
+    def possible_events(self) -> DecoderEventMappingType:
+        return {CPT_THEGRAPH: {
+            HistoryEventType.STAKING: {
+                HistoryEventSubType.DEPOSIT_ASSET: EventCategory.DEPOSIT,
+                HistoryEventSubType.REMOVE_ASSET: EventCategory.WITHDRAW,
+            },
+            HistoryEventType.INFORMATIONAL: {
+                HistoryEventSubType.NONE: EventCategory.INFORMATIONAL,
+            },
+        }}
+
+    def addresses_to_decoders(self) -> dict[ChecksumEvmAddress, tuple[Any, ...]]:
+        return {
+            CONTRACT_THEGRAPH_STAKING: (self._decode_delegator_staking,),
+        }
+
+    def _decode_delegator_staking(self, context: DecoderContext) -> DecodingOutput:
+        if context.tx_log.topics[0] == TOPIC_STAKE_DELEGATED:
+            return self._decode_stake_delegated(context)
+        elif context.tx_log.topics[0] == TOPIC_STAKE_DELEGATED_LOCKED:
+            return self._decode_stake_locked(context)
+        elif context.tx_log.topics[0] == TOPIC_STAKE_DELEGATED_WITHDRAWN:
+            return self._decode_stake_withdrawn(context)
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_stake_delegated(self, context: DecoderContext) -> DecodingOutput:
+        deposit_event, burn_event = None, None
+        delegator = hex_or_bytes_to_address(context.tx_log.topics[2])
+        if delegator is None or self.base.is_tracked(delegator) is False:
+            return DEFAULT_DECODING_OUTPUT
+        indexer = hex_or_bytes_to_address(context.tx_log.topics[1])
```

Comment:
Use newline after if. Helps readability

```suggestion
        indexer = hex_or_bytes_to_address(context.tx_log.topics[1])

```